From ada1c30d27e3b079b54a857094197aad4bdcac3f Mon Sep 17 00:00:00 2001
From: Victor Toso <me@victortoso.com>
Date: Fri, 29 Sep 2017 18:43:24 +0200
Subject: [PATCH 40/55] vdagent: move file xfer finalize to a function

This patch creates vdagent_finalize_file_xfer() to finalize and stop
file xfer. Moving this code to a function removes some duplication.

Signed-off-by: Victor Toso <victortoso@redhat.com>
Acked-by: Frediano Ziglio <fziglio@redhat.com>
---
 src/vdagent/vdagent.c | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/src/vdagent/vdagent.c b/src/vdagent/vdagent.c
index ac507c4..7a2a220 100644
--- a/src/vdagent/vdagent.c
+++ b/src/vdagent/vdagent.c
@@ -109,6 +109,15 @@ static gboolean vdagent_init_file_xfer(void)
     return (vdagent_file_xfers != NULL);
 }
 
+static gboolean vdagent_finalize_file_xfer(void)
+{
+    if (vdagent_file_xfers == NULL)
+        return FALSE;
+
+    g_clear_pointer(&vdagent_file_xfers, vdagent_file_xfers_destroy);
+    return TRUE;
+}
+
 static void daemon_read_complete(struct udscs_connection **connp,
     struct udscs_message_header *header, uint8_t *data)
 {
@@ -160,10 +169,7 @@ static void daemon_read_complete(struct udscs_connection **connp,
         if (debug)
             syslog(LOG_DEBUG, "Disabling file-xfers");
 
-        if (vdagent_file_xfers != NULL) {
-            vdagent_file_xfers_destroy(vdagent_file_xfers);
-            vdagent_file_xfers = NULL;
-        }
+        vdagent_finalize_file_xfer();
         break;
     case VDAGENTD_AUDIO_VOLUME_SYNC: {
         VDAgentAudioVolumeSync *avs = (VDAgentAudioVolumeSync *)data;
@@ -185,9 +191,7 @@ static void daemon_read_complete(struct udscs_connection **connp,
         break;
     case VDAGENTD_CLIENT_DISCONNECTED:
         vdagent_x11_client_disconnected(x11);
-        if (vdagent_file_xfers != NULL) {
-            vdagent_file_xfers_destroy(vdagent_file_xfers);
-            vdagent_file_xfers = NULL;
+        if (vdagent_finalize_file_xfer()) {
             vdagent_init_file_xfer();
         }
         break;
@@ -400,9 +404,7 @@ reconnect:
         udscs_client_handle_fds(&client, &readfds, &writefds);
     }
 
-    if (vdagent_file_xfers != NULL) {
-        vdagent_file_xfers_destroy(vdagent_file_xfers);
-    }
+    vdagent_finalize_file_xfer();
     vdagent_x11_destroy(x11, client == NULL);
     udscs_destroy_connection(&client);
     if (!quit && do_daemonize)
-- 
2.16.1

